taxi/views.py

```diff
@@ -34,77 +48,187 @@ class ManufacturerListView(LoginRequiredMixin, generic.ListView):
 
+        return context
+
+    def get_queryset(self):
+        form = ManufacturerSearchForm(self.request.GET)
+        if form.is_valid():
+            return self.queryset.filter(
+                name__icontains=form.cleaned_data["name"]
+            )
+
+        return self.queryset
+
+
+class ManufacturerDetailView(LoginRequiredMixin, generic.DetailView):
+    model = Manufacturer
+
+
+@method_decorator(staff_member_required(login_url="/"), name="dispatch")
 class ManufacturerCreateView(LoginRequiredMixin, generic.CreateView):
     model = Manufacturer
     fields = "__all__"
     success_url = reverse_lazy("taxi:manufacturer-list")
 
 
-class ManufacturerUpdateView(LoginRequiredMixin, generic.UpdateView):
+@method_decorator(staff_member_required(login_url="/"), name="dispatch")
+class ManufacturerUpdateView(
+    LoginRequiredMixin, SweetifySuccessMixin, generic.UpdateView
+):
     model = Manufacturer
     fields = "__all__"
     success_url = reverse_lazy("taxi:manufacturer-list")
+    success_message = "Manufacturer successfully update"
 
 
-class ManufacturerDeleteView(LoginRequiredMixin, generic.DeleteView):
+@method_decorator(staff_member_required(login_url="/"), name="dispatch")
+class ManufacturerDeleteView(generic.DeleteView):
     model = Manufacturer
     success_url = reverse_lazy("taxi:manufacturer-list")
 
 
 class CarListView(LoginRequiredMixin, generic.ListView):
     model = Car
-    paginate_by = 5
+    paginate_by = 4
     queryset = Car.objects.all().select_related("manufacturer")
 
+    def get_context_data(self, *, object_list=None, **kwargs):
+        context = super(CarListView, self).get_context_data(**kwargs)
+        model = self.request.GET.get("model", "")
+        context["search_form"] = CarSearchForm(initial={"model": model})
 
-class CarDetailView(LoginRequiredMixin, generic.DetailView):
+        return context
+
+    def get_queryset(self):
+        form = CarSearchForm(self.request.GET)
+        if form.is_valid():
+            return self.queryset.filter(
+                model__icontains=form.cleaned_data["model"]
+            )
+        return self.queryset
+
+
+class CarDetailView(LoginRequiredMixin, FormMixin, generic.DetailView):
     model = Car
+    form_class = CarCommentForm
+
+    def post(self, request, *args, **kwargs):
+        form = self.get_form()
+        if form.is_valid:
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
 
+    def get_success_url(self, **kwargs):
+        return reverse_lazy(
+            "taxi:car-detail", kwargs={"pk": self.get_object().id}
+        )
 
+    def form_valid(self, form):
+        self.object = form.save(commit=False)
+        self.object.car = self.get_object()
+        self.object.driver = self.request.user
+        self.object.save()
+        return super().form_valid(form)
+
+
+@method_decorator(staff_member_required(login_url="/"), name="dispatch")
```

Comment:
Why do you use staff_member_required? You can use LoginRequiredMixin instead of this decorator